netcore/NetCore.Session/NetCoreSessionState.cs

```diff
@@ -0,0 +1,377 @@
+            byteIndex += sizeof(long);
+
+            // Append the value
+            Array.Copy(_value, 0, byteArray, byteIndex, _value.Length);
+            return byteArray;
+        }
+
+        public void FromByteArray(byte[] data)
+        {
+            int byteIndex = 0;
+
+            // Extract the LastAccessTimeUtc
+            LastAccessTimeUtc = DateTime.FromBinary(BitConverter.ToInt64(data, byteIndex));
+            byteIndex += sizeof(long);
+
+            // Extract the ExpirationTimeUtc
+            ExpirationTimeUtc = DateTime.FromBinary(BitConverter.ToInt64(data, byteIndex));
+            byteIndex += sizeof(long);
+
+            // Extract the SpanUntilStale
+            SpanUntilStale = TimeSpan.FromTicks(BitConverter.ToInt64(data, byteIndex));
+            byteIndex += sizeof(long);
+
+            // Extract the value
+            Value = new byte[data.Length - byteIndex];
+            Array.Copy(data, byteIndex, _value, 0, data.Length - byteIndex);
+        }
+    }
+
+    public class GeodeSessionStateCache : GeodeNativeObject, IDistributedCache
+    {
+        private readonly IGeodeCache _cache;
+        private ILogger<GeodeSessionStateCache> _logger;
+        private static Region _region;
+        private string _regionName;
+        private readonly SemaphoreSlim _connectLock = new SemaphoreSlim(initialCount: 1, maxCount: 1);
+
+        public GeodeSessionStateCache(IOptions<GeodeSessionStateCacheOptions> optionsAccessor) {
+
+            var host = optionsAccessor.Value.Host;
+            var port = optionsAccessor.Value.Port;
+            _regionName = optionsAccessor.Value.RegionName;
+
+            _cache = CacheFactory.Create()
+                .SetProperty("log-level", "none")
+                .CreateCache();
+
+            _cache.PoolManager
+                .CreatePoolFactory()
+                .AddLocator(host, port)
+                .CreatePool("pool");
+
+            var regionFactory = _cache.CreateRegionFactory(RegionShortcut.Proxy);
+            _region = regionFactory.CreateRegion(_regionName);
+        }
+
+        // Returns the SessionStateValue for key, or null if key doesn't exist
+        public GeodeSessionStateValue GetValueForKey(string key)
+        {
+            byte[] cacheValue = _region.GetByteArray(key);
+
+            if (cacheValue != null)
+            {
+                return new GeodeSessionStateValue(cacheValue);
+            }
+            else
+                return null;
+        }
+
+        public byte[] Get(string key)
+        {
+            if (key == null)
+            {
+                throw new ArgumentNullException(nameof(key));
+            }
+
+            Connect();
+
+            // Check for nonexistent key
+            GeodeSessionStateValue ssValue = GetValueForKey(key);
+            if (ssValue == null)
+                return null;
+
+            // Check for expired key
+            DateTime nowUtc = DateTime.UtcNow;
```

Comment:
Use `var` throughout rather than redundantly specifying the type. Consistency with the client source is key.

Reply:
Good catch.